### ISSUE TITLE
trainingg end with nan loss

### ISSUE THREAD

**Issue by Fly-dream12** [fanq15/FewX#4]
Thanks for your code! When I try to run it on my dataset, the error output is:
  File "FewX-master/fewx/data/dataset_mapper.py", line 234, in generate_support
                support_id= self.support_df.loc[
                (self.support_df['category_id'] == query_cls) &
                 (~self.support_df['image_id'].isin(used_image_id))
                & (~self.support_df['id'].isin(used_id_ls))
                , 'id'].sample(random_state=id).tolist()[0]

 1)Is there anything wrong with generating pickle file?
and if i change the code to : support_id= self.support_df.loc[
                (self.support_df['category_id'] == query_cls)  , 'id'].sample(random_state=id).tolist()[0]
this error happens:
detectron2/detectron2/modeling/proposal_generator/proposal_utils.py", line 95, in find_top_rpn_proposals
    "Predicted boxes or scores contain Inf/NaN. Training has diverged."
FloatingPointError: Predicted boxes or scores contain Inf/NaN. Training has diverged.
2) Could you give some suggestions?

**Comment by fanq15** [comment]
You can try to reduce both the `support_shot` and `support_way` to 1. It should be ok. You can not remove the `(~self.support_df['image_id'].isin(used_image_id))
& (~self.support_df['id'].isin(used_id_ls))`, because we have to make sure the image and instance are not used in both the query and support image.

**Comment by fanq15** [comment]
If you use your own dataset, you can try to reduce the learning rate. A large learning rate sometimes may cause the training to divergence.

**Comment by Fly-dream12** [comment]
I have reduced the learning rate to 0.0001, but it still went nan loss.
In fewx/modeling/fsod/fsod_rcnn.py, 158 line the support waymust be 2 ?

**Comment by fanq15** [comment]
Sorry, you can not change the `support_way`. You can reference the paper for details. The contrastive training is very important for the performance, especially on the MS COCO dataset. Do you use your own dataset?

**Comment by fanq15** [comment]
Currently, the code only supports the 2-way contrastive training.

**Comment by Fly-dream12** [comment]
Yes. I have tried as you have suggested, but the first error still happens.
I wonder if I have made the wrong pickle file. 
Can you give a detailed description of generating pickle file?

**Comment by fanq15** [comment]
You should make sure that your dataset format is in the COCO format, especially the `bbox` format in your json file.

**Comment by fanq15** [comment]
You can print `self.support_df` to debug. You have to make sure there are at least 2 images for each class when the `support_shot` is 1.

**Comment by WeitaoVan** [comment]
Hi @Fly-dream12 @fanq15 , I met the same error:

`query_cls = self.support_df.loc[self.support_df['id']==id, 'category_id'].tolist()[0] # they share the same category_id and image_id
IndexError: list index out of range`

I followed the README to generate train_support_df.pkl, I guess there is something wrong with the generated pickle file, have you found the solution? Thanks!
